Nikhil Naik/ListDesignApp/components/BottomDrawer.tsx

```diff
@@ -80,6 +82,24 @@ const BottomDrawer = ({ open }: any) => {
       _data();
     }
   }, [open]);
+  useEffect(() => {
+    if (search.length > 1) {
+      let temp = [];
+      const filter = [];
+      const newData = data.filter((item, idx) => {
+        temp = item.version;
+        temp = temp.filter((items) => {
+          return items.toLowerCase().indexOf(search.toLowerCase()) != -1;
+        });
+        if (temp.length > 0) {
+          filter.push({ makeName: item.makeName, version: temp });
+        }
+      });
+      setFilterData(filter);
+    } else {
+      setFilterData(data);
+    }
```

Comment:
instead of re-rendering the entire data , try to modify the current data. with this useEffect on every key press this makes the entire search part and all of its components such as accordion, and Accordionanimation and accordionItem to get re-rendered.